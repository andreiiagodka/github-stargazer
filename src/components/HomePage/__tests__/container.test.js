import { shallow } from 'enzyme'
import configureStore from 'redux-mock-store'

import HomePageContainer from '..'

describe('<HomePageContainer />', () => {
  const mockStore = configureStore([])

  it('renders SpinnerComponent', () => {
    const initialState = {
      fetchRepositories: { repositories: [], loading: true, error: false }
    }
    const store = mockStore(initialState)

    const wrapper = shallow(<HomePageContainer store={store} />)
    const container = wrapper.dive().dive().dive()

    expect(container).toMatchSnapshot()
  })

  it('renders ErrorComponent', () => {
    const initialState = {
      fetchRepositories: { repositories: [], loading: false, error: true }
    }
    const store = mockStore(initialState)

    const wrapper = shallow(<HomePageContainer store={store} />)
    const container = wrapper.dive().dive().dive()

    expect(container).toMatchSnapshot()
  })

  it('renders LayoutComponent', () => {
    const initialState = {
      fetchRepositories: { repositories: [], loading: false, error: false }
    }
    const store = mockStore(initialState)

    const wrapper = shallow(<HomePageContainer store={store} />)
    const container = wrapper.dive().dive().dive()

    expect(container).toMatchSnapshot()
  })
})
