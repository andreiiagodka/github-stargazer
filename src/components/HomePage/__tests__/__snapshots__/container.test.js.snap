// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomePageContainer /> renders ErrorComponent 1`] = `
<Alert
  className="text-center"
  closeLabel="Close alert"
  show={true}
  transition={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "defaultProps": Object {
        "appear": false,
        "in": false,
        "mountOnEnter": false,
        "timeout": 300,
        "unmountOnExit": false,
      },
      "render": [Function],
    }
  }
  variant="danger"
>
  Something went wrong!
</Alert>
`;

exports[`<HomePageContainer /> renders LayoutComponent 1`] = `
<Fragment>
  <CardHeader>
    <HeaderComponent />
  </CardHeader>
  <CardBody>
    <ContentComponent
      repositories={Array []}
    />
  </CardBody>
</Fragment>
`;

exports[`<HomePageContainer /> renders SpinnerComponent 1`] = `
<div
  className="Loader"
>
  Loading ...
</div>
`;
